pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go

```diff
@@ -99,17 +100,17 @@ func (c *SyntheticServiceEntryController) List(typ, namespace string) (out []mod
 				out = append(out, *config)
 			}
 		}
-		return out, nil
+	} else {
+		byNamespace := c.configStore[namespace]
+		for _, config := range byNamespace {
+			out = append(out, *config)
+		}
 	}
 
-	byNamespace, ok := c.configStore[namespace]
-	if !ok {
-		return nil, nil
-	}
+	sort.Slice(out, func(i, j int) bool {
```

Comment:
Just curious why we need to sort by config name?

Reply:
To have a determined output, as we do for other configs